main/externalData/inventory/processor/index.ts

```diff
@@ -2,13 +2,35 @@ import log from 'electron-log'
 
 import store from '../../../store'
 
-const setInventory = (account: string, inventory: Inventory) => {
-  store.setInventory(account.toLowerCase(), inventory)
+const storeApi = {
+  getInventory(account: string) {
+    return (store('main.inventory', account.toLowerCase()) || {}) as Inventory
+  },
+  setInventory(account: string, inventory: Inventory) {
+    store.setInventory(account.toLowerCase(), inventory)
+  },
+  setInventoryAsset(account: string, collection: string, tokenId: string, item: InventoryAsset) {
+    store.setInventoryAsset(account.toLowerCase(), collection, tokenId, item)
+  }
 }
 
-const updateItems = (account: string, items: InventoryAsset[]) => {
-  const inventory = store('main.inventory', account.toLowerCase()) as Inventory
-  items.forEach((item) => {
+export const updateCollections = (account: string, inventory: Inventory) => {
+  const existingInventory = storeApi.getInventory(account)
+
+  for (const [contractAddress, { items }] of Object.entries(inventory)) {
+    for (const tokenId of Object.keys(items)) {
+      const tokenData = existingInventory[contractAddress]?.items[tokenId]
```

Comment:
is it possible for this to be undefined? even if it is, we should fail outside this loop